### PR TITLE
[Docs] Update cross-compilation guide for OSX

### PR DESCRIPTION
**Pull request by yokofly**
1. Sync the cross-compilation code for use in the CI action docker:
   - Use `--depth=1` during cloning to reduce clone times.
   - Update to a newer version of the macOS SDK. see:
https://github.com/ClickHouse/ClickHouse/blob/eb3a7caa74748a5c9f97f3d740cefb16f9a4bae2/docker/packager/binary/Dockerfile#L82

2. Intro cross-compilation support for aarch64 OSX. The key change involves replacing the `x86_64` phase with `aarch64`, which is the current configuration we use.

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by yokofly** [comment]
below is difference about current 2 arch diff in cross compile
only replace the `x86_64` -> `aarch64` , and create a folder for SDK download.
https://www.diffchecker.com/aEum7CGb/

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 46703bf7b9231dc3d0eb09285b628da5b661d87b with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56199/46703bf7b9231dc3d0eb09285b628da5b661d87b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6727293245">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56199/46703bf7b9231dc3d0eb09285b628da5b661d87b/ci_running.html">❌ failure</a></td></tr>
<tbody>
</table>

File: docs/en/development/build-cross-osx.md [hunk 1]
```diff
@@ -60,3 +70,46 @@ ninja
 ```
 
 The resulting binary will have a Mach-O executable format and can’t be run on Linux.
+
+# ARM (aarch64) macOS:
```
Comment (rschu1ze) [inline docs/en/development/build-cross-osx.md:74]: Thanks!

There is quite a bit of copypasta here (which will make future maintenance more difficult). Could we maybe only keep the description for x86 and add comments à la "To build for ARM, replace `x86` by `aarch64`".

Comment (yokofly) [reply docs/en/development/build-cross-osx.md:74]: sure

**Review by rschu1ze — APPROVED** [review]

**Comment by yokofly** [comment]
thanks! I forgot to ask for the label for this PR. hahaha

**Pull request merged**